cfme/base/credential.py

```diff
@@ -0,0 +1,178 @@
+# -*- coding: utf-8 -*-
+from cfme.web_ui import FileInput, Input, Radio, form_buttons
+from cfme.web_ui.tabstrip import TabStripForm
+from utils.pretty import Pretty
+from utils.update import Updateable
+
+
+class Credential(Pretty, Updateable):
+    """
+    A class to fill in credentials
+
+    Args:
+        principal: Something
+        secret: Something
+        verify_secret: Something
+    """
+    pretty_attrs = ['principal', 'secret']
+
+    def __init__(self, principal, secret, verify_secret=None, domain=None, **ignore):
+        self.principal = principal
+        self.secret = secret
+        self.verify_secret = verify_secret
+        self.domain = domain
+
+    def __getattribute__(self, attr):
+        if attr == 'verify_secret':
+            if object.__getattribute__(self, 'verify_secret') is None:
+                return object.__getattribute__(self, 'secret')
+            else:
+                return object.__getattribute__(self, 'verify_secret')
+
+        elif attr == 'principal':
+            domain = object.__getattribute__(self, 'domain')
+            principal = object.__getattribute__(self, 'principal')
+            return r'{}\{}'.format(domain, principal) if domain else principal
+        else:
+            return super(Credential, self).__getattribute__(attr)
+
+    def as_dict(self):
```

Comment:
we don't need __getattribute__ there. ronny, already brought my attention to it. I've created a card about that.